### PR TITLE
GIDS: Add the ability to upload a single tab excel file #14735

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
https://github.com/department-of-veterans-affairs/va.gov-team/issues/14735

- Removed using SmarterCSV code, the code and gem will be removed in another PR
- Allows for upload of xls, xlsx, txt, csv files through roo
- Allows for uploads of minorly malformed xls, xlsx files 
- Forced ruby/rails to require certain classes because wasn't loading correctly

## Testing done
- Uploaded a csv, txt, xls, xlsx file
- Uploaded files with headers without underscores
- Uploaded bad files and viewed new message order when failing

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: config/settings.yml [hunk 1]
```diff
@@ -6,6 +6,13 @@ active_record:
 archiver:
   archive: true
 
+upload:
```
Comment (zurbergram) [inline config/settings.yml:9]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types

File: lib/roo_helper/shared.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module RooHelper
+  include Common
+  def self.included(base)
+    base.extend Common::Shared
+    base.extend Loader
+    base.extend CsvHelper::Exporter
```
Comment (zurbergram) [inline lib/roo_helper/shared.rb:8]: Going to create the exporter in a separate PR

File: lib/csv_helper/loader.rb [hunk 3]
```diff
@@ -2,6 +2,10 @@
```
Comment (zurbergram) [inline lib/csv_helper/loader.rb:2]: SmarterCSV will be removed in a separate PR

File: spec/models/shared_examples/shared_examples_for_loadable.rb [hunk 4]
```diff
@@ -39,25 +38,100 @@
...
+    #     expect(upload.col_sep).to eq('|')
+    #   end
+    #
+    #   it 'raises error when neither comma or pipe are found' do
+    #     first_line = 'a/b\c'
+    #     upload = create :upload
+    #     error_message = 'Unable to determine column separators, valid separators equal "|" and ","'
+    #     expect { upload.send(:set_col_sep, first_line) }.to raise_error(StandardError, error_message)
+    #   end
+    # end
```
Comment (shawkey-daniel) [inline spec/models/shared_examples/shared_examples_for_loadable.rb:135]: Can we remove the commented code?

**Review by shawkey-daniel — APPROVED** [review]
LGTM

File: spec/models/shared_examples/shared_examples_for_loadable.rb (hunk 4 above)
Comment (zurbergram) [reply spec/models/shared_examples/shared_examples_for_loadable.rb:135]: That was to remind myself to write new unit tests that cover these tests since the code was removed. 
I'll get those unit tests written

Comment (zurbergram) [reply spec/models/shared_examples/shared_examples_for_loadable.rb]: Removed and rewritten

**Review by thilton-oddball — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]
LGTM
